### PR TITLE
⭐️ Adding support for fetching the code Owner Approval from Protected Branch (Gitlab)

### PR DESCRIPTION
**Pull request by HRouhani**
Screenshot from 2024-07-22 09-53-58

### CODE REVIEW

File: providers/gitlab/resources/gitlab.lr [hunk 1]
```diff
@@ -121,11 +121,13 @@ gitlab.project.approvalSettings @defaults("approvalsBeforeMerge requirePasswordT
 
 
 // GitLab protected branch
-gitlab.project.repository.protectedBranch @defaults("name allowForcePush") {
+gitlab.project.protectedBranch @defaults("name allowForcePush") {
```
Comment (chris-rock) [inline providers/gitlab/resources/gitlab.lr]: i think that resource should also be private

File: providers/gitlab/resources/gitlab.lr [hunk 2]
```diff
@@ -121,11 +121,13 @@ gitlab.project.approvalSettings @defaults("approvalsBeforeMerge requirePasswordT
...
   // Branch name
   name string
   // Whether force push is allowed
   allowForcePush bool
   // Whether this is the default branch
   defaultBranch bool
+  // Whether code owner approval required
+  CodeOwnerApproval bool
```
Comment (chris-rock) [inline providers/gitlab/resources/gitlab.lr]: we always start with lower case

File: providers/gitlab/resources/gitlab.go [hunk 3]
```diff
@@ -228,12 +228,13 @@ func (p *mqlGitlabProject) protectedBranches() ([]interface{}, error) {
 		isDefaultBranch := branch.Name == defaultBranch
 
 		branchSettings := map[string]*llx.RawData{
-			"name":           llx.StringData(branch.Name),
-			"allowForcePush": llx.BoolData(branch.AllowForcePush),
-			"defaultBranch":  llx.BoolData(isDefaultBranch),
+			"name":              llx.StringData(branch.Name),
+			"allowForcePush":    llx.BoolData(branch.AllowForcePush),
+			"defaultBranch":     llx.BoolData(isDefaultBranch),
+			"CodeOwnerApproval": llx.BoolData(branch.CodeOwnerApprovalRequired),
```
Comment (chris-rock) [inline providers/gitlab/resources/gitlab.go]: this needs to be lower case at the beginning

**Review by chris-rock — COMMENTED** [review]
Great addition @HRouhani I added a comment about the resource name

File: providers/gitlab/resources/gitlab.lr (hunk 1 above)
Comment (HRouhani) [reply providers/gitlab/resources/gitlab.lr]: Thanks, did it for all of the new added!

File: providers/gitlab/resources/gitlab.lr (hunk 2 above)
Comment (HRouhani) [reply providers/gitlab/resources/gitlab.lr]: Thanks, missed that one! corrected

File: providers/gitlab/resources/gitlab.go (hunk 3 above)
Comment (HRouhani) [reply providers/gitlab/resources/gitlab.go]: done , Thanks

File: providers/gitlab/resources/gitlab.lr.manifest.yaml [hunk 4]
```diff
@@ -92,10 +93,20 @@ resources:
...
+    is_private: true
     min_mondoo_version: 9.0.0
+  gitlab.project.protectedBranch:
+    fields:
+      allowForcePush: {}
+      codeOwnerApproval: {}
+      defaultBranch: {}
+      name: {}
+    is_private: true
+    min_mondoo_version: latest
```
Comment (tas50) [inline providers/gitlab/resources/gitlab.lr.manifest.yaml]:
```suggestion
    min_mondoo_version: 9.0.0
```

Comment (HRouhani) [reply providers/gitlab/resources/gitlab.lr.manifest.yaml]: done. Thanks

File: providers/gitlab/resources/gitlab.go [hunk 5]
```diff
@@ -228,12 +228,13 @@ func (p *mqlGitlabProject) protectedBranches() ([]interface{}, error) {
...
-			"allowForcePush": llx.BoolData(branch.AllowForcePush),
-			"defaultBranch":  llx.BoolData(isDefaultBranch),
+			"name":              llx.StringData(branch.Name),
+			"allowForcePush":    llx.BoolData(branch.AllowForcePush),
+			"defaultBranch":     llx.BoolData(isDefaultBranch),
+			"codeOwnerApproval": llx.BoolData(branch.CodeOwnerApprovalRequired),
 		}
 
-		mqlProtectedBranch, err := CreateResource(p.MqlRuntime, "gitlab.project.repository.protectedBranch", branchSettings)
+		mqlProtectedBranch, err := CreateResource(p.MqlRuntime, "gitlab.project.protectedBranch", branchSettings)
```
Comment (vjeffrey) [inline providers/gitlab/resources/gitlab.go:237]: it looks like the id function for this still needs to be updated for the new naming

Comment (HRouhani) [reply providers/gitlab/resources/gitlab.go:237]: Thanks, Its correct:

func (g *mqlGitlabProjectProtectedBranch) id() (string, error) {
	return g.Name.Data, nil
}

**Review by vjeffrey — APPROVED** [review]
thank u!

**Pull request merged**